### ISSUE TITLE
codeql fails when using a go.work file

### ISSUE THREAD

**Issue by mxpaspa** [github/codeql#14235]
Codeql fails to scan a go project initialized with a go.work file

<img width="1040" alt="image" src="https://github.com/github/codeql/assets/18662680/78084d75-8a2c-48ca-b08e-88a698355262">

Is there a suggested workaround to get codeql to run with a go.work project? 

Also, has this pr been implemented somewhere else? https://github.com/github/codeql/pull/12012

**Comment by smowton** [comment]
No, it hasn't. The workaround for now would be to replace the autobuild step with a manual build command that builds your `go` code without using `go.work`, such as using `replace` directives in `go.mod` as was the recommended practice before `go.work` was introduced. I will note that `go.work` usages is picking up; we intend to implement support, and your report will make it more likely to happen sooner!

**Comment by mbg** [comment]
`go.work` files should now be supported as of CodeQL 2.16.4. If you are encountering any further issues with `go.work` files, please open a new issue!
